hw/ip/spi_host/doc/_index.md

```diff
@@ -0,0 +1,633 @@
+3. Holds the bus in a high resistance state for some number of "dummy" clock cycles (neither side transmits), or
+4. Receives information from the target device.
+
+{{< wavejson >}}
+{signal: [
+  {name: "clk_core_i", wave: "p.............................."},
+  {name: "sck (CPOL=0)", wave: "0.1010101010101010101010101010."},
+  {name: "cs_n", wave: "10............................1"},
+  {name: "sd[0]", wave: "22.2.2.2.2.2.2.2.2.2.z...2.2.x.", data: ["","cmd7", "cmd6", "cmd5", "cmd4", "cmd3", "cmd2", "cmd1", "cmd0",
+                                                             "out7", "out3", "in7", "in3" ]},
+  {name: "sd[1]", wave: "z................2.2.z...2.2.x.", data: ["out6", "out2", "in6", "in2"]},
+  {name: "sd[2]", wave: "z................2.2.z...2.2.x.", data: ["out5", "out1", "in5", "in1"]},
+  {name: "sd[3]", wave: "z................2.2.z...2.2.x.", data: ["out4", "out0", "in4", "in0"]},
+  {name: "Transaction phase:", wave: "x2...............3...4...5...x.", data: ['TX1', 'TXn', 'Dummy','RX'] },
+  {name: "Sampling Event (FULLCYC=0)", wave: "1.........................H.H.."},
+  {name: "Sampling Event (FULLCYC=1)", wave: "1..........................H.H."},
+],
+  head: {
+  text: "Example Quad SPI transaction: 1 byte TX (Single), 1 byte (Quad), 2 dummy cycles and 1 RX byte with CPHA=0"
+  },
+}
+
+{{< /wavejson >}}
+
+For even faster transfer rates, some flash chips support double transfer rate (DTR) variations to the SPI protocol wherein the device receives and transmits fresh data on *both* the leading and trailing edge.
+This IP only supports single transfer rate (STR), *not* DTR.
+A preliminary investigation of DTR transfer mode suggests that proper support for setup and hold times in this mode may require a level of sub-cycle timing control which is not currently planned for this IP.
+
+# Theory of Operations
+
+## SPI_HOST IP Command Interface
+
+Issuing a command generally consists of the following steps:
+1. Configure the IP to be compatible with the target peripheral, using the {{< regref "CONFIG" >}} register,
+2. Load the instructions and data to be transmitted into the TXFIFO via the {{< regref "TXDATA" >}} register,
+3. Issue a 32-bit command to the {{< regref "COMMAND" >}} register with the following information,
+    - The speed to use for this particular command (Standard, Dual, or Quad)
+    - The three bit chip-select ID of the peripheral to target
+    - The number of bytes to transmit at *reduced* data rate, using `sd[0]` only (Dual or Quad rate *only*)
+        - This field is specifically for Dual or Quad commands where the first instruction must be sent serially on `sd[0]`
+    - The number of bytes to transmit at full data rate
+        - In Standard mode this controls the number of bytes transmitted and the number of bytes received, since the bus is full-duplex in Standard mode.
+    - The number of dummy `sck` cycles to wait before reading data (Dual or Quad rate *only*)
```

Comment:
there are actually non dual / quad commands that also have dummy cycles. 
Does it make sense to not restrict these?

Reply:
I don't think it matters if they are dummy (high-Z), or just explicitly padded.
I think @rswarbrick pointed out a similar restriction in the json file in that the Standard-mode is currently full-duplex.  However I'm changing that in the next commit.

Reply:
that might not be a safe assumption to make... we've definitely run into cases where either a host or device doesn't let go of the lines during dummy cycles, and that causes conflict on the line, which is very damaging long term.

Reply:
Okay, well for standard... no device should *ever* be grabbing the other sides lines.   sd[0] is should *always* controlled by the host, and sd[1] should *always* controlled by the device.   Though I can imagine without a real standard for SPI, device vendors can technically do whatever they want.  Sounds like a real nuisance though.

Nonetheless... the full-duplex standard mode operation is going away in the next commit. (at least as a default)

While I'm at it, I'll also add a flag to tri-state sd[0] if there is no actual data to send on it.

Reply:
Hi @tjaychen, 
   This will be addressed as of the last commit (3f90caeb9439dec510a503a44aeeed92ce1bf55c ).  Half-duplex is now the default for Standard SPI commands.  AND when it is assumed that the remote side does not care what the data being transmitted is sd[0] can be either tied low, or left high-z (as in Dual or Quad mode) depending on the new HIGHZ flag.

Reply:
With that, can we consider this technical point resolved?

Reply:
sounds good to me.